### ISSUE TITLE
Inconsistency with Request constructor and cf object

### ISSUE THREAD

**Issue by jdavis** [cloudflare/miniflare#535]
There is an inconsistency with how Miniflare handles a new creation of a request object along with the `cf` object when supplied as an init option. This inconsistency doesn't happen within the Cloudflare Worker runtime environment however.

Simplified example code showing the issue:

```
const newRequest = new Request("http://test.com");
const cfRequest = new Request(newRequest, { cf: { cacheKey: "key", cacheTtl: 100, , cacheEverything: true });
console.log(cfRequest.cf);
```

It'll result in different output depending on if it runs within Miniflare or Cloudflare's Worker.

Cloudflare worker output: `{ cacheKey: "key", cacheTtl: 100, , cacheEverything: true}`
Miniflare locally: `undefined`

The bug appears to be this [line of code](https://github.com/cloudflare/miniflare/blame/master/packages/core/src/standards/http.ts#L427) in the `Request` constructor:
```
const cf = input instanceof Request ? input.#cf : init?.cf;
```

I'm invoking my fetch handler within a unit test and creating new Requests in the handler it'll never propagate the `cf` object because `input instanceof Request` is always true.

Am I overlooking something here or doing something I shouldn't be with Miniflare? Or is this actually a bug like I suspect?

Any suggestions on getting around this? As it is right now, I can't do any unit test validation on caching with code that works within Cloudflare Worker's runtime because the `cf` object is never propagating to the new request locally in tests.

**Comment by birkensubstack** [comment]
It seems like [this example](https://developers.cloudflare.com/workers/examples/modify-request-property/) with the following code:

```
  // Best practice is to always use the original request to construct the new request
  // to clone all the attributes. Applying the URL also requires a constructor
  // since once a Request has been constructed, its URL is immutable.
  const newRequest = new Request(url.toString(), new Request(request, newRequestInit));
```

Would agree with you. The `init` should override the `input` for all these properties.

What is the fix, something like:

```const cf = init?.cf ?? (input instanceof Request ? input.#cf : undefined);```

**Comment by jdavis** [comment]
Yeah, I think that's probably the simple fix. I just don't know the impact of doing that and if that would break other people's code. I'm curious what @mrbbot thinks

**Comment by jmaroeder** [comment]
Just noticed this error myself in my own unit tests.

One workaround that may apply for _some_ situations is that `fetch` will correctly combine the request and the `init` object, so if you're able to call `fetch` directly you can avoid this oversight. In fact, I uncovered this bug while in the midst of a refactor to pass around a `Request` instead of the `fetch` result
